lib/__tests__/factory.test.ts

```diff
@@ -96,6 +96,40 @@ describe('factory.createList', () => {
     expect(users.every(u => u.name === 'Bill')).toBeTruthy();
     expect(onCreateFn).toHaveBeenCalledTimes(2);
   });
+
+  it('calls onBulkCreate function if bulkCreate value passed true in options', async () => {
+    const onBulkCreateFn = jest.fn(async users => {
+      users.every((user: { name: string; }) => user.name = 'Bill')
+      return Promise.resolve(users);
+    });
+
+    const bulkUserFactory = userFactory.onBulkCreate(onBulkCreateFn);
+
+    const promise = bulkUserFactory.createList(2, { name: 'susan' }, { bulkCreate: true });
+    expect(promise).toBeInstanceOf(Promise);
+
+    const users = await promise;
+
+    expect(users.every(u => u.name === 'Bill')).toBeTruthy();
+    expect(onBulkCreateFn).toHaveBeenCalledTimes(1);
+  });
+
+  it('does not calls onBulkCreate function if bulkCreate value passed false in options', async () => {
```

Comment:
I think this gets tested along with `calls onCreate for each item`, no need to duplicate the test.   You can add `expect(onBulkCreateFn).toHaveBeenCalledTimes(0);` in that test case.

Reply:
@dhirendrx in that case we cannot `expect(onBulkCreateFn) ` till that we should have definition for it